lib/collection/src/shards/queue_proxy_shard.rs

```diff
@@ -308,42 +320,42 @@ struct Inner {
     /// See `set_wal_keep_from()` and `UpdateHandler::wal_keep_from` for more details.
     /// Defaults to `u64::MAX` to allow acknowledging all confirmed versions.
     wal_keep_from: Arc<AtomicU64>,
+    /// Optional progression tracker.
+    progress: Option<Arc<ParkingMutex<TransferTaskProgress>>>,
 }
 
 impl Inner {
     pub fn new(
         wrapped_shard: LocalShard,
         remote_shard: RemoteShard,
         wal_keep_from: Arc<AtomicU64>,
+        progress: Option<Arc<ParkingMutex<TransferTaskProgress>>>,
     ) -> Self {
         let start_from = wrapped_shard.wal.wal.lock().last_index() + 1;
-
-        let shard = Self {
+        Self::new_from_version(
             wrapped_shard,
             remote_shard,
-            transfer_from: start_from.into(),
-            update_lock: Default::default(),
             wal_keep_from,
-        };
-
-        // Keep all new WAL entries so we don't truncate them off when we still need to transfer
-        shard.set_wal_keep_from(Some(start_from));
```

Comment:
Is this line completely gone?

Reply:
No. It is now called in the `new_from_version` method.